app/src/main/java/com/fish/monsters/features/settings/SettingsViewModel.kt

```diff
@@ -0,0 +1,89 @@
+package com.fish.monsters.features.settings
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.compose.runtime.State
+import androidx.core.os.LocaleListCompat
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.fish.monsters.common.models.ui.Language
+import com.fish.monsters.common.utils.MusicManager
+import com.fish.monsters.common.utils.settings.SettingsGlobalState
+import com.fish.monsters.common.utils.settings.SettingsManager
+import com.fish.monsters.core.database.FishDatabase
+import com.fish.monsters.core.database.dao.SettingsDao
+import com.fish.monsters.core.database.entities.Settings
+import com.fish.monsters.features.settings.ui.SettingsEvent
+import kotlinx.coroutines.launch
+
+class SettingsViewModel(
+    private val settingsManager: SettingsManager,
+    private val musicManager: MusicManager,
+    private val settingsDao: SettingsDao,
+    private val fishDb: FishDatabase
+) : ViewModel() {
+    val settingsGlobalState: State<SettingsGlobalState> = settingsManager.state
```

Comment:
Tutaj wyjątkowo polegamy na `state` z innej klasy jako że to co zmienamy na tym screenie musimy przekazać na `scope` całej aplikacji. Zazwyczaj w `viewModel`ach state holder będzie to wyglądać tak jak w [SettingsManagerze.kt](https://github.com/arix2000/FishMonsters/pull/4/files#diff-7971ede44b95d47297980c344c100d6a5aeecc4d21c664d2e8413e77ff1f8e01)

Sample:
```kotlin
private val _state = mutableStateOf(MyStateHolder())
    val state: State<MyStateHolder> = _state
```